test/integration/puppetlabs/services/legacy_routes/legacy_routes_test.clj

```diff
@@ -32,6 +32,9 @@
 (deftest ^:integration legacy-routes
   (testing "The legacy web routing service properly handles old routes."
     (bootstrap/with-puppetserver-running-with-mock-jrubies
+     "MOCK.TODO: discuss this one.  Currently all this is testing is that the
```

Comment:
Yeah, that's a good point.  Definitely a good argument to be made that the original tests for this were already insufficient.

I think the primary thing we need to cover is ensuring that the request made to the old route is eventually funneled down as the new route into the Ruby Puppet layer.  To that end, we *could* still handle this via mocking if we had the mock `.handleRequest` method just echo back the request map into the response body and have these tests inspect the response body to see that each request has been mapped to the appropriate new indirector route by the legacy routes service.  I suspect you might still think that's insufficient, though.

If you think we need to do an end-to-end test down through the indirector terminii, I can go ahead and convert this back to a non-mock `with-puppetserver-running` and add some extra assertions against the response body content for each.